src/bentoml/server.py

```diff
@@ -431,5 +432,6 @@ def get_client(self) -> GrpcClient | None:
             GrpcClient.wait_until_server_ready(
                 host=self.host, port=self.port, timeout=self.timeout
             )
-            self._client = GrpcClient.from_url(f"{self.host}:{self.port}")
+            client_host = "localhost" if self.host in ["0.0.0.0", "::"] else self.host
```

Comment:
```suggestion
            client_host = "localhost" if self.host in ["0.0.0.0", "::"] else self.host
```
gRPC actually support ipv6 by default. so `::3000` should work

Reply:
There are a bunch of reasons that wouldn't work, but importantly this is more about the fact that `::` and `0.0.0.0` are not valid addresses.